core/types/qi_tx.go

```diff
@@ -121,6 +121,7 @@ func CalculateQiTxGas(transaction *Transaction) uint64 {
 	if transaction.Type() != QiTxType {
 		panic("CalculateQiTxGas called on a transaction that is not a Qi transaction")
 	}
+	// TODO: This should check for ETXs (and conversion) and calculate expected gas for those as well
```

Comment:
this todo relevent ?

Reply:
https://github.com/dominant-strategies/go-quai/issues/1643